src/main/java/net/neoforged/neoform/runtime/utils/ToolCoordinate.java

```diff
@@ -0,0 +1,51 @@
+package net.neoforged.neoform.runtime.utils;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * We use this enumeration and property file so that the Gradle build-script can easily
+ * look up the versions we use and expose them in a separate Gradle dependency list.
+ * It also makes it far easier to integrate a renovate bot.
+ */
+public enum ToolCoordinate {
+    JAVA_SOURCE_TRANSFORMER,
+    DIFF_PATCH;
+
+    private static final Properties VERSIONS;
+
+    private MavenCoordinate version;
+
+    static {
+        try (var in = ToolCoordinate.class.getResourceAsStream("Tools.properties")) {
```

Comment:
Please please please please please use lowercase `tools.properties`

Also, generally you use `/<file>`